File: src/RoomBooking.Server/wwwroot/app/core/domain/roomReservation.ts
export class RoomReservation {
    Id: string;
    ClientName: string;
    StartDate: Date;
    EndDate: Date;
    Room: string;
    RoomId: string;
    Hotel: string;
    HotelId: string;

    constructor(id: string,
        clientname: string,
        startdate: Date,
        enddate: Date,
        room: string,
        roomid: string,
        hotel: string,
        hotelid: string
    ) {
        this.Id = id;
        this.ClientName = clientname;
        this.StartDate = startdate;
        this.EndDate = enddate;
        this.Room = room;
        this.Hotel = hotel;
        this.RoomId = roomid;
    }
}

File: src/RoomBooking.Shared/Entities/RoomReservation.cs
using System;
using RoomBooking.Shared.Entities.Abstract;

namespace RoomBooking.Shared.Entities
{
    public class RoomReservation : IEntityBase
    {
        public string ClientName { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public Room Room { get; set; }
        public Guid Id { get; set; }
    }
}